### PR TITLE
exec/util: Introduce File struct and rewrite WriteFile accordingly

### PR DESCRIPTION
**Pull request by vcaputo**

### CODE REVIEW

File: exec/util/unit.go [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+
+	"github.com/coreos/ignition/config"
+)
+
+func FileFromUnit(root string, unit config.Unit) *File {
+	return &File{
+		Path:     filepath.Join(root, SystemdUnitsPath(), string(unit.Name)),
+		Contents: unit.Contents,
+		Mode:     DefaultFilePermissions,
+		Uid:      0,
```
Comment (crawford) [inline exec/util/unit.go]: Why not omit these lines?

File: exec/util/file.go [hunk 2]
```diff
@@ -23,18 +22,23 @@ import (
...
+}
 
-func WriteFile(filename, contents string) error {
+func (f *File) Write() error {
 	var err error
 
-	dir := filepath.Dir(filename)
-	if err := EnsureDirectoryExists(dir); err != nil {
+	dir := filepath.Dir(f.Path)
+	if err := os.MkdirAll(dir, DefaultDirectoryPermissions); err != nil {
```
Comment (crawford) [inline exec/util/file.go]: Will this fail the same way as `EnsureDirectoryExists`?

File: exec/util/unit.go (hunk 1 above)
Comment (vcaputo) [inline exec/util/unit.go]: was just being explicit

File: exec/util/file.go (hunk 2 above)
Comment (vcaputo) [inline exec/util/file.go]: The previous stuff was unnecessary, MkdirAll() does the right thing:

>     MkdirAll creates a directory named path, along with any necessary
    parents, and returns nil, or else returns an error. The permission bits
    perm are used for all directories that MkdirAll creates. If path is
    already a directory, MkdirAll does nothing and returns nil.

**Comment by crawford** [comment]
As discussed in person, can you move the File structure to the config package and leave the Write() implementation in its current location?

File: config/config.go [hunk 3]
```diff
@@ -19,7 +19,8 @@ import (
 )
 
 type Config struct {
-	Systemd Systemd
+	Systemd    Systemd
+	WriteFiles []File `yaml:"write_files"`
```
Comment (crawford) [inline config/config.go]: Please omit this for now. This will live in a different part of the global config.

**Comment by crawford** [comment]
Can you just prune the config_test.go and Config changes? Otherwise, LGTM.

**Comment by vcaputo** [comment]
Why don't we just leave them for now since they're the only thing exercising the changes currently?

**Comment by crawford** [comment]
@vcaputo I'd rather omit all of that since that code will be changed soon anyway (I don't think we are going to use YAML). It was there as a rough framework but was meant to be swapped out.

**Pull request merged**